src/graphql/account.rs

```diff
@@ -420,6 +373,125 @@ impl AccountMutation {
     }
 }
 
+/// This asynchronous function performs several checks and operations to authenticate a user based
+/// on the provided username and password, enforce IP restrictions, manage concurrent sessions,
+/// and update or verify passwords.
+///
+/// # Returns
+///
+/// Returns a `Result` containing an `AuthPayload` if the sign-in is successful, or an error
+/// message (`String`) if any of the checks fail or an error occurs during the process.
+///
+/// # Errors
+///
+/// This function returns an error in the following situations:
+///
+/// - If the username is not found in the account map, the function returns an "incorrect
+///   username  or password" error.
+/// - If the client IP cannot be retrieved and the account has IP restrictions, an "unable to
+///   retrieve client IP" error is returned.
+/// - If the client's IP address is not in the `allow_access_from` list for the account,
+///   an "access denied from this IP" error is returned.
+/// - If the number of active sessions exceeds the account's `max_parallel_sessions` limit,
+///   a "maximum parallel sessions exceeded" error is returned.
+/// - If the provided password does not match the stored password, an "incorrect username or
+///   password" error is returned.
+/// - If the `is_update_password` flag is set to true and the new password is the same as the
+///   current password, a "password is the same as the previous one" error is returned.
+/// - If the account's `last_signin_time` is `None` and a password change is required, an
+///   "a password change is required to proceed" error is returned.
+/// - If there is an error creating a token, updating the last sign-in time, or storing the
+///   account or token data, the respective error messages are returned.
```

Comment:
Internal methods do not require the Rust documentation. Explanation suffices, if needed.

Reply:
When responding to a reviewer's request for changes, it's better to do so both when you first review the request and when you've completed the requested changes . If I had to choose one, I would prefer the latter.

Reply:
Yes, I see.
The function that left document has been removed, making the long document above completely unnecessary.